lib/model/folder.go

```diff
@@ -450,6 +451,51 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 
 		batch.append(res.File)
 		changes++
+
+		err := snap.WithPathsMatchingBlocksHash(res.File.BlocksHash, func(path string) bool {
+			select {
+			case <-f.ctx.Done():
+				return false
+			default:
+			}
+
+			cf, ok := snap.Get(protocol.LocalDeviceID, path)
+			if !ok {
+				return true
+			}
+			if !bytes.Equal(cf.BlocksHash, res.File.BlocksHash) {
+				l.Debugf("Mismatching block map list hashes: got %x expected %x", cf.BlocksHash, res.File.BlocksHash)
+				return true
+			}
+			if cf.IsDeleted() || cf.IsInvalid() || cf.IsIgnored() || cf.IsUnsupported() || cf.IsDirectory() || cf.IsSymlink() {
+				l.Debugf("Found something of unexpected type in block map list: %s", cf)
+				return true
+			}
+			if f.ignores.Match(cf.Name).IsIgnored() {
```

Comment:
This check isn't necessary, the scanner completely ignores ignored items (what a sentence), i.e. doesn't return them.

Reply:
Sure, but this is a random filename that we got from the blocklist. It could have been ignored after it was added to it?

Reply:
Right of course, I messed up the `cf`s and `res`'.